sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessorBase.cs

```diff
@@ -0,0 +1,729 @@
+        ///
+        private Task RenewOwnershipAsync()
+        {
+            IEnumerable<PartitionOwnership> ownershipToRenew = InstanceOwnership.Values
+                .Select(ownership => new PartitionOwnership
+                (
+                    ownership.FullyQualifiedNamespace,
+                    ownership.EventHubName,
+                    ownership.ConsumerGroup,
+                    ownership.OwnerIdentifier,
+                    ownership.PartitionId,
+                    ownership.Offset,
+                    ownership.SequenceNumber,
+                    DateTimeOffset.UtcNow,
+                    ownership.ETag
+                ));
+
+            // We cannot rely on the ownership returned by ClaimOwnershipAsync to update our InstanceOwnership dictionary.
+            // If the user issues a checkpoint update, the associated ownership will have its eTag updated as well, so we
+            // will fail in claiming it here, but this instance still owns it.
+
+            return ClaimOwnershipAsync(ownershipToRenew);
+        }
+
+        /// <summary>
+        ///   Starts running a task responsible for receiving and processing events in the context of a specified partition.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition the task is associated with.  Events will be read only from this partition.</param>
+        /// <param name="startingPosition">The position within the partition where the task should begin reading events.</param>
+        /// <param name="maximumReceiveWaitTime">The maximum amount of time to wait to for an event to be available before emitting an empty item; if <c>null</c>, empty items will not be published.</param>
+        /// <param name="retryOptions">The set of options to use for determining whether a failed operation should be retried and, if so, the amount of time to wait between retry attempts.</param>
+        /// <param name="trackLastEnqueuedEventInformation">Indicates whether or not the task should request information on the last enqueued event on the partition associated with a given event, and track that information as events are received.</param>
+        ///
+        /// <returns>The running task that is currently receiving and processing events in the context of the specified partition.</returns>
+        ///
+        protected virtual Task RunPartitionProcessingAsync(string partitionId,
+                                                           EventPosition startingPosition,
+                                                           TimeSpan? maximumReceiveWaitTime,
+                                                           RetryOptions retryOptions,
+                                                           bool trackLastEnqueuedEventInformation) => Task.Run(async () =>
+            {
+                // TODO: let the user include a cancellation token?  Why would they need to stop the processing?
```

Comment:
We should;  on general principal `protected` == `public` and we should always allow for a cancellation token on public async.